### ISSUE TITLE
Support other platforms

### ISSUE THREAD

**Issue by slashmo** [slashmo/install-swift#1]
`install-swift` only supports running on Ubuntu (`18.04` & `20.04`). Supporting other runner os's like macOS and Windows would make the action more useful.

## Platforms to support

- [x] #5
- [ ] #8
- [ ] #9
- [ ] #10

**Comment by adam-fowler** [comment]
amazonlinux2 and centos7/8 would be good as well

**Comment by slashmo** [comment]
> amazonlinux2 and centos7/8 would be good as well
> @adam-fowler 

Thank you, that's a great suggestion! How would you run your CI on GitHub Actions using these distributions? Using the Swift Docker images? In that case, what would be the benefit of installing a specific version of Swift inside the image, if you could have selected this version as part of the image tag already?

**Comment by adam-fowler** [comment]
I guess you are right there. Forgot GH actions only have Ubuntu runners. I guess if someone has their own runner they can have whatever version of Linux installed on it. Maybe the action should at least recognise this, and download the correct version of swift

**Comment by slashmo** [comment]
> Maybe the action should at least recognise this, and download the correct version of swift
> @adam-fowler 

Good point, then let's try and support all platforms where both Swift and the GitHub runner tool runs on.
